src/ReferenceList.vue

```diff
@@ -8,8 +8,34 @@
 <script>
 import ReferenceWidget from './ReferenceWidget.vue'
 import axios from '@nextcloud/axios'
+import { getBuilder } from '@nextcloud/browser-storage'
 import { generateOcsUrl } from '@nextcloud/router'
 import { URL_PATTERN } from './helpers.js'
+import sha1 from 'js-sha1'
+
+const browserStorage = getBuilder('vue-richtext').clearOnLogout().build()
+
+function getStorageKey(text, limit) {
+	return `references-${sha1(text)}-${limit}`
+}
+
+function getReferences(text, limit) {
+	const references = browserStorage.getItem(getStorageKey(text, limit))
+	if (references) {
+		const referenceObject = JSON.parse(references)
+		if (referenceObject.expiry > Date.now()) {
+			return referenceObject
+		}
+		browserStorage.removeItem(getStorageKey(text, limit))
+	}
+	return null
+}
+
+function setReferences(text, limit, references) {
+	const expiry = Date.now() + 3600 * 1000
```

Comment:
This is 1h in js, right?🤔

The i guess it's good. Because e.g. github labels change and that should reflect at some point

Reply:
The server-side cache is also one hour
https://github.com/nextcloud/server/blob/0431ad8a02225ae822b8c28db95a78d01bc7ac4c/lib/private/Collaboration/Reference/ReferenceManager.php#L40
https://github.com/nextcloud/server/blob/0431ad8a02225ae822b8c28db95a78d01bc7ac4c/lib/private/Collaboration/Reference/ReferenceManager.php#L101

Reply:
Yes, both are set to one hour.